public/app/plugins/panel/heatmap-new/hovertabs/ExemplarTab.tsx

```diff
@@ -0,0 +1,145 @@
+import React from 'react';
+
+import { ArrayVector, DataFrame, dateTimeFormat, systemDateFormats, TimeRange, TimeZone } from '@grafana/data';
+
+import { DataHoverView } from '../components/DataHoverView';
+import { BucketLayout, getHeatmapFields } from '../fields';
+import { HeatmapHoverProps, HeatmapLayerHover } from '../types';
+
+interface HeatmapLayerOptions {
+  timeZone: TimeZone;
+  timeRange: TimeRange;
+}
+
+export const ExemplarTab = ({
+  heatmapData,
+  index,
+  options,
+}: HeatmapHoverProps<HeatmapLayerOptions>): HeatmapLayerHover => {
+  if (!heatmapData?.heatmap) {
+    return {
+      name: 'Exemplar',
+    };
+  }
+  const [xField, yField, countField] = getHeatmapFields(heatmapData?.heatmap!);
+  const mapping: number[] | null = heatmapData.exemplarsMappings?.lookup[index!]!;
+  const count: number = mapping?.length ?? 0;
+
+  if (
+    xField &&
+    yField &&
+    countField &&
+    typeof count === 'number' &&
+    count > 0 &&
+    typeof index !== 'undefined' &&
+    index >= 0
+  ) {
+    const yValueIdx = index % heatmapData?.yBucketCount! ?? 0;
+
+    const yMinIdx = heatmapData.yLayout === BucketLayout.le ? yValueIdx - 1 : yValueIdx;
+    const yMaxIdx = heatmapData.yLayout === BucketLayout.le ? yValueIdx : yValueIdx + 1;
+
+    const xMin: number = xField.values.get(index);
+    const xMax: number = xMin + heatmapData.xBucketSize!;
+    const yMin: number = yField.values.get(yMinIdx);
+    const yMax: number = yField.values.get(yMaxIdx);
+
+    if (count === 0) {
+      return {
+        name: 'Exemplar',
+      };
+    }
+
+    const summaryData: DataFrame = {
```

Comment:
This *may* be necessary, but it seems a bit complex.  IIUC, creates a new frame with a single row for the exemplar.  I think there is a way to do support this with fewer hoops

Reply:
@ryantxu Unless we wrote a control that specifically put in summary data, I don't see how we could remove this. The dataframe is only to use the generic `DataHoverView` control. That's literally it. We could write a "SummaryData" control that does the same thing. It might appear as fewer hoops, because we don't create a new dataframe, and could pass the numbers in directly. Thoughts?